matchzoo/preprocessors.py

```diff
@@ -228,3 +228,70 @@ def fit(self, tokens: list, ngram: int=3):
         """
         _, input_dim = self._create_n_letters(tokens, ngram)
         self._state = {'input_dim': input_dim}
+
+
+class Vocabulary(StatefulProcessorUnit):
+    """
+    Vocabulary class.
+
+    Examples:
+        >>> vocab = Vocabulary()
+        >>> vocab.fit(['A', 'B', 'C', 'D', 'E'])
+        >>> term_index = vocab.state['term_index']
+        >>> term_index  # doctest: +SKIP
+        {'E': 1, 'C': 2, 'D': 3, 'A': 4, 'B': 5}
+        >>> index_term = vocab.state['index_term']
+        >>> index_term  # doctest: +SKIP
+        {1: 'C', 2: 'A', 3: 'E', 4: 'B', 5: 'D'}
+
+        >>> term_index['out-of-vocabulary-term']
+        0
+        >>> index_term[0]
+        ''
+        >>> index_term[42]
+        Traceback (most recent call last):
+            ...
+        KeyError: 42
+
+        >>> a_index = term_index['A']
+        >>> c_index = term_index['C']
+        >>> vocab.transform(['C', 'A', 'C']) == [c_index, a_index, c_index]
+        True
+        >>> vocab.transform(['C', 'A', 'OOV']) == [c_index, a_index, 0]
+        True
+
+        >>> indices = vocab.transform('ABCDDZZZ')
+        >>> ''.join(vocab.state['index_term'][i] for i in indices)
+        'ABCDD'
+
+    """
+
+    class IndexTerm(dict):
+        """Map index to term."""
+
+        def __missing__(self, key):
+            """Map out-of-vocabulary indices to empty string."""
+            if key == 0:
+                return ''
```

Comment:
I think the best thing to return is `<oov>` instead of an empty string.

Reply:
Not necessarily. `0` could be the result of padding.

Reply:
I thought about separating oov and padding, but kept it simple.

Reply:
okay